FreeRTOS-Plus/Demo/Device_Shadow_Windows_Simulator/Device_Shadow_Demo/DemoTasks/shadow_demo_helpers.c

```diff
@@ -0,0 +1,934 @@
+                xConnectInfo.pPassword = democonfigCLIENT_PASSWORD;
+                xConnectInfo.passwordLength = ( uint16_t ) strlen( democonfigCLIENT_PASSWORD );
+            #else
+                xConnectInfo.pUserName = AWS_IOT_METRICS_STRING;
+                xConnectInfo.userNameLength = AWS_IOT_METRICS_STRING_LENGTH;
+                /* Password for authentication is not used. */
+                xConnectInfo.pPassword = NULL;
+                xConnectInfo.passwordLength = 0U;
+            #endif /* ifdef democonfigCLIENT_USERNAME */
+
+            /* Send MQTT CONNECT packet to broker. */
+            xMQTTStatus = MQTT_Connect( pxMqttContext,
+                                        &xConnectInfo,
+                                        NULL,
+                                        mqttexampleCONNACK_RECV_TIMEOUT_MS,
+                                        &sessionPresent );
+
+            if( xMQTTStatus != MQTTSuccess )
+            {
+                returnStatus = pdFAIL;
+                LogError( ( "Connection with MQTT broker failed with status %u.", xMQTTStatus ) );
+            }
+            else
+            {
+                LogInfo( ( "MQTT connection successfully established with broker.\n\n" ) );
+            }
+        }
+
+        if( returnStatus == pdFAIL )
+        {
+            /* Keep a flag for indicating if MQTT session is established. This
+             * flag will mark that an MQTT DISCONNECT has to be sent at the end
+             * of the demo even if there are intermediate failures. */
+            mqttSessionEstablished = true;
+        }
+
+        if( returnStatus == pdFAIL )
+        {
+            /* Check if session is present and if there are any outgoing publishes
+             * that need to resend. This is only valid if the broker is
+             * re-establishing a session which was already present. */
+            if( sessionPresent == true )
+            {
+                LogInfo( ( "An MQTT session with broker is re-established. "
+                           "Resending unacked publishes." ) );
+
+                /* Handle all the resend of publish messages. */
+                returnStatus = xHandlePublishResend( &mqttContext );
+            }
+            else
+            {
+                LogInfo( ( "A clean MQTT connection is established."
+                           " Cleaning up all the stored outgoing publishes.\n\n" ) );
+
+                /* Clean up the outgoing publishes waiting for ack as this new
+                 * connection doesn't re-establish an existing session. */
+                vCleanupOutgoingPublishes();
+            }
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+BaseType_t xDisconnectMqttSession( void )
+{
+    MQTTStatus_t xMQTTStatus = MQTTSuccess;
+    BaseType_t returnStatus = pdTRUE;
+    TlsTransportStatus_t xNetworkStatus;
```

Comment:
This is never used in the function. Please remove.

Reply:
Thanks~!